lib/context2d.js

```diff
@@ -125,14 +125,13 @@ Object.defineProperty(Context2d.prototype, 'currentTransform', {
  */
 
 Context2d.prototype.__defineSetter__('fillStyle', function(val){
-  if (!val) return;
-  if ('CanvasGradient' == val.constructor.name
-    || 'CanvasPattern' == val.constructor.name) {
+  if (val instanceof CanvasGradient
+    || val instanceof CanvasPattern) {
     this.lastFillStyle = val;
     this._setFillPattern(val);
-  } else if ('string' == typeof val) {
+  } else {
     this.lastFillStyle = undefined;
-    this._setFillColor(val);
+    this._setFillColor(val + '');
```

Comment:
`val + ''` uses `val.valueOf()` instead of `toString` if there's a `valueOf` method. Whatever  Chrome does internally uses `toString` (e.g. `this._setFillColor(String(val))`.

```js
> let x = {toString: () => "#0000ff", valueOf: () => "#00ff00"}
> ctx.fillStyle = x
> ctx.fillStyle
"#0000ff"
```

Reply:
also, I think `String(val)` makes the intent more clear ☺️